src/components/templates/PosterHall/PosterHall.tsx

```diff
@@ -19,10 +23,21 @@ export interface PosterHallProps {
 }
 
 export const PosterHall: React.FC<PosterHallProps> = ({ venue }) => {
+  const venueId = useVenueId();
+  const { isRelatedVenuesLoading, relatedVenueIds } = useRelatedVenues({
+    currentVenueId: venueId,
+  });
+  const { events: relatedVenueEvents } = useVenueEvents({
+    venueIds: relatedVenueIds,
+  });
```

Comment:
Looking at the query config for `useConnectPosterRelatedVenues` in `src/hooks/posters.ts`:

```typescript
{
  collection: "venues",
  where: [
    ["parentId", "==", posterHallId],
    ["template", "in", [VenueTemplate.audience, VenueTemplate.jazzbar]],
  ],
  storeAs: "posterRelatedVenues",
}
```

It's getting the `audience`/`jazzbar` venues that are direct children of the `posterHallId`, which is passed in as props from `usePosters`, and so in this file is set to `venue.id`

We're already passing `venueId` into `useRelatedVenues` here, so we can probably just `.filter` the `relatedVenues` array in a `useMemo` or similar to find venues where:

```typescript
const fooRelatedVenues = relatedVenues.filter((relatedVenue) => relatedVenue.parentId === venue.id && [VenueTemplate.audience, VenueTemplate.jazzbar].includes(relatedVenue.template))
```

This wouldn't live update, but I expect that isn't needed here as the venues are unlikely to be changing.

We would then get the `id`s from just these `fooRelatedVenues` (needs better name; this would likely be done with `.map`, and would also need to be done in a `useMemo`), and pass that to `useVenueEvents`, which would then only fetch the events for these venues, rather than fetching all of them; which you could then filter to get just the live ones in the `useInterval`, and then set that result to the `useState`, etc.

This would then allow us to get rid of all of the `useFirestoreConnect`/selector/firestore types/etc that were added in this PR and simplify things right down here.

Reply:
Thanks for all the help with this — I managed to implement what you described, and seems to finally be working smoothly again.

Reply:
Glad to hear it :)